app/controllers/foreman_tasks/concerns/environments_extension.rb

```diff
@@ -0,0 +1,24 @@
+module ForemanTasks
+  module Concerns
+    module EnvironmentsExtension
+      extend ActiveSupport::Concern
+
+      included do
+        alias_method_chain :obsolete_and_new, :dynflow
+      end
+
+      def obsolete_and_new_with_dynflow
+        task = ForemanTasks.async_task(::Actions::Foreman::Puppetclass::Import,
+                                       params)
+
+        notice _("Added import task to queue, it will be run shortly")
```

Comment:
I have note realized the link is not shown properly in the notification window, so it just renders in page escaped. Sry about that - we would need a better infra in UI to make it better. So the version is ok.